### PR TITLE
Use optimize=True in einsum

### PR DESCRIPTION
**Pull request by schneiderfelipe**
## Description

As explained in #85, part [1f of the tutorial](https://github.com/psi4/psi4numpy/blob/a894b1c4246c183a4cc88e2d8cd6ad275821cbbc/Tutorials/01_Psi4NumPy-Basics/1f_tensor-manipulation.ipynb) discourages the usage of the `optimize=True` flag of `einsum`  due to the numpy version of its introduction (1.12) being too recent. But that's a long time ago. This PR tries to solve this and introduces the flag in every tutorial `einsum` is used.

## What are your new additions? Please provide a brief list.
* **Changes**
  - [x] The appropriate tutorial now explains the flag and compares timings
  - [x] All tutorials make use of the `optimize=True` flag of `einsum`
  - [x] Some `FutureWarning`s were corrected in the DFT tutorials

## Status
- [x] Click when ready for review-and-merge

### CODE REVIEW

File: Tutorials/01_Psi4NumPy-Basics/1f_tensor-manipulation.ipynb [hunk 1]
```diff
@@ -272,31 +292,33 @@
...
+      "Now Numpy's optimized transformation...\n",
+      "...optimized transformation complete in 0.002 seconds.\n",
+      "\n",
       "\n",
       "Starting Numpy's N^5 transformation with dot...\n",
       "...transformation complete in 0.001 seconds.\n",
       "\n",
       "Allclose: True\n",
-      "N^5 22712.25 faster than N^8 algorithm!\n"
+      "N^5 1.54 faster than N^8 algorithm!\n"
```
[the remark this answers is not in the document]
Comment (schneiderfelipe) [reply Tutorials/01_Psi4NumPy-Basics/1f_tensor-manipulation.ipynb]: It looks like I redefined the variable `n8_time`. My fault.

File: Tutorials/12_MD/12a_basics.ipynb [hunk 2]
```diff
@@ -40,11 +38,802 @@
...
-   "execution_count": null,
-   "metadata": {
-    "collapsed": true
-   },
-   "outputs": [],
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
```
[the remark this answers is not in the document]
Comment (schneiderfelipe) [reply Tutorials/12_MD/12a_basics.ipynb]: I reverted all MD tutorials back to the original files.

**Comment by schneiderfelipe** [comment]
> Overall LGTM. Please do revert the MD tutorials so they they correctly show their 3-d structures.

Let me know if it's OK now.

**Comment by dgasmith** [comment]
Thanks! Looks good.

**Pull request merged**